### PR TITLE
Final bit of code for basic auth 🎤 ⬇️ 

### PR DESCRIPTION
**Pull request by krachwal**
Added basic auth so that bugs in dependencies can be resolved in private.

### CODE REVIEW

File: react-frontend/src/components/Form.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ function Form() {
 		setWaitingForResponse(true);
 		setNoConvergence(false);
 
-		let websocket = new WebSocket('ws://localhost/data');
```
Comment (krachwal) [inline react-frontend/src/components/Form.tsx:75]: plugging in actual host

File: react-frontend/src/components/Charts.tsx [hunk 2]
```diff
@@ -336,17 +336,21 @@ function Charts({ a_n, b_n, limit, symbol, convergesTo, deltaData, toggleDisplay
...
-						Delta is a measure of the irrationality of a number (read more about it{' '}
-						<a href="https://www.ramanujanmachine.com/the-mathematics-of-polynomial-continued-fractions/irrationality-testing/">
-							here
-						</a>
-						). The given Polynomial Continued Fraction produces the following finite-depth
-						estimations for Delta:
-					</p>
-					<ScatterPlot id="delta_chart" data={deltaData} />
-				</div>
+				{deltaData && deltaData?.length > 0 ? (
```
Comment (krachwal) [inline react-frontend/src/components/Charts.tsx:339]: not showing the chart if the data fails to load on the backend (they have a bug that sometimes crashes divide by zero)

File: Dockerfile [hunk 3]
```diff
@@ -10,6 +10,17 @@ WORKDIR /srv/ramanujan-machine-web-portal
 ARG wolfram_app_id
 ENV WOLFRAM_APP_ID=$wolfram_app_id
 
+# set temporary basic auth credentials
+ARG admin_user
+ENV ADMIN_USER=$admin_user
+ARG admin_password
+ENV ADMIN_PASSWORD=$admin_password
+RUN echo "ADMIN_USER=${ADMIN_USER}" >> .env
```
[the remark this answers is not in the document]
Comment (davidbrownell) [reply Dockerfile:18]: But, now that I think about it, I'm not so sure :)

[the remark this answers is not in the document]
Comment (krachwal) [reply Dockerfile:18]: This is just to keep the general public out until they fix their logic. It doesn't have to be super secure.

[the remark this answers is not in the document]
Comment (davidbrownell) [reply Dockerfile:18]: I verified that it is included in the image's history:  
  
(this is a test image that I created)

[the remark this answers is not in the document]
Comment (davidbrownell) [reply Dockerfile:18]: You can work around this by creating file(s) with the username/password on the host, copying them to the image, using those files to modify the `.env` file, and then removing the file(s) on the image.  
  
Or, given that it is temporary, add a comment in the dockerfile explaining how this is a security vulnerability and should be changed as soon as possible.

[the remark this answers is not in the document]
Comment (krachwal) [reply Dockerfile:18]: I think I'll go with the comment. All of this happens on a runner. The runner gets wiped between executions. If someone gets onto the ec2 instance to see the layers, that is a much bigger problem than them seeing the web UI.

[the remark this answers is not in the document]
Comment (davidbrownell) [reply Dockerfile:18]: Sounds good. Maybe include in the comment that they should never push the image to a public image repository.

**Review by davidbrownell — APPROVED** [review]

**Pull request merged**